### PR TITLE
cpu: Initial import for the Freescale K20 MCU 

### PR DESCRIPTION
**Pull request by DipSwitch**
This is the initial import for the Freescale K20 family, which is used on the Teensy board series.

### CODE REVIEW

File: cpu/k20/Makefile.include [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# define path to kinetis module, which is needed for this CPU
+export KINETIS_COMMON = $(RIOTCPU)/kinetis_common/
+# CPU depends on the kinetis module, so include it
+include $(KINETIS_COMMON)Makefile.include
+
+#export the CPU model and architecture
+MODEL = $(shell echo $(CPU_MODEL)|tr 'a-z' 'A-Z')
+export CFLAGS += -DCPU_MODEL_$(MODEL)
+ARCH = $(shell echo $(CPU_ARCH) | tr 'a-z-' 'A-Z_')
+export CFLAGS += -DCPU_ARCH_$(ARCH)
```
Comment (haukepetersen) [inline cpu/k20/Makefile.include]: these defiles are being also exported by the `Makefile.include.cortexm_common`, so I think they could be dropped here, right?

File: cpu/k20/include/mk20dx128.c [hunk 2]
```diff
@@ -0,0 +1,1000 @@
+/* Teensyduino Core Library
```
Comment (haukepetersen) [inline cpu/k20/include/mk20dx128.c]: a .c file in the include path? Seems like this file is redundant with the `startup.c` and also with a lot of things defined in the cortexm_common part.

**Comment by haukepetersen** [comment]
Cool, @thomaseichinger gave me his teensy board today, hopefully I can test this soon!

**Comment by DipSwitch** [comment]
Currently it stops working, when all thread go to sleep. This probably has
to do with invalid PLL / FLL config, watchdog or UART / IRQ configuration.
My teensy is already at the office (have thin wires there), but I didn't
have time yet to connect wires to the SWD interface and bypass the build in
bootloader. :) But tomorrow I'll have time ^_^

File: cpu/k20/Makefile.include (hunk 1 above)
Comment (DipSwitch) [inline cpu/k20/Makefile.include]: This is copied from the K60 implementation. I guess this is done for the linker scripts which are all capital for the Kinetis cores.

**Comment by gebart** [comment]
There are two leftover K60 files in the ldscripts directory in this PR.

File: cpu/k20/cpu.c [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ */
+void cpu_init(void)
+{
+    /* initialize the Cortex-M core */
+    cortexm_init();
+    /* Check that we are running on the CPU that this code was built for */
+    check_running_cpu_revision();
+}
+
+static void check_running_cpu_revision(void)
```
Comment (jfischer-phytec-iot) [inline cpu/k20/cpu.c]: please drop check_running_cpu_revision if  you do not use it

File: cpu/k20/include/mk20dx128.c (hunk 2 above)
Comment (jfischer-phytec-iot) [inline cpu/k20/include/mk20dx128.c]: :+1:

File: cpu/kinetis_common/mcg.c [hunk 4]
```diff
@@ -153,7 +153,7 @@ static void kinetis_mcg_enable_osc(void)
         MCG->C2 = (uint8_t)(MCG_C2_RANGE0(0));
     }
 
-    OSC0->CR = (uint8_t)(OSC_CR_ERCLKEN_MASK | OSC_CR_EREFSTEN_MASK
+    OSC->CR = (uint8_t)(OSC_CR_ERCLKEN_MASK | OSC_CR_EREFSTEN_MASK
```
Comment (jfischer-phytec-iot) [inline cpu/kinetis_common/mcg.c]: Please do not change cpu/kinetis_common/mcg.c  here (if there is something wrong in kinetis_common then open a PR for it)

**Comment by OlegHahm** [comment]
@DipSwitch, any progress?

**Comment by kYc0o** [comment]
WIP -> postponed.

**Comment by miri64** [comment]
Postponed due to Feature Freeze.

**Comment by OlegHahm** [comment]
@DipSwitch, please indicate if you will continue to work on this or if we should close this.

**Comment by gebart** [comment]
This seem abandoned, reopen if you want to continue this PR in the future.

**Pull request closed without merging**